homeassistant/components/bittrex/__init__.py

```diff
@@ -0,0 +1,93 @@
+"""Gather the market details from Bittrex."""
+import logging
+from typing import Dict
+
+from aiobittrexapi import Bittrex
+from aiobittrexapi.errors import (
+    BittrexApiError,
+    BittrexInvalidAuthentication,
+    BittrexResponseError,
+)
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONF_API_SECRET, CONF_MARKETS, DOMAIN, SCAN_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.All(
+            cv.deprecated(CONF_API_KEY),
+            cv.deprecated(CONF_API_SECRET),
+            cv.deprecated(CONF_MARKETS),
+            vol.Schema(
+                {
+                    vol.Optional(CONF_API_KEY): cv.string,
+                    vol.Optional(CONF_API_SECRET): cv.string,
+                    vol.Optional(CONF_MARKETS): cv.string,
+                }
+            ),
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass: HomeAssistant, config: Dict) -> bool:
+    """Set up the component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Bittrex from a config entry."""
+    api_key = entry.data[CONF_API_KEY]
+    api_secret = entry.data[CONF_API_SECRET]
+    symbols = entry.data[CONF_MARKETS]
+
+    coordinator = BittrexDataUpdateCoordinator(hass, api_key, api_secret, symbols)
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    hass.async_create_task(
+        hass.config_entries.async_forward_entry_setup(entry, "sensor")
+    )
+
+    return True
+
+
+class BittrexDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to get the latest data from Bittrex."""
+
+    def __init__(self, hass, api_key, api_secret, symbols):
+        """Initialize the data object."""
+        self.bittrex = Bittrex(api_key, api_secret)
+        self.symbols = symbols
+        self.hass = hass
```

Comment:
I haven't written an integration using coordinators, but if they are like entities then I suspect super().__init__ will save the hass field for you.

Reply:
Hmm, I think it really needs it:
![image](https://user-images.githubusercontent.com/14926452/103474770-5839eb00-4da7-11eb-939f-24549655de4e.png)

I also searched for `(DataUpdateCoordinator):` in the components folder and I see several components doing it like this. 

Let me know if I'm missing something!

Reply:
Unfortunately 'what other integrations do' is a bad way to decide if something is needed -- many integrations set all kinds of bad examples! The source of truth is the source for DataUpdateCoordinator: `DataUpdateCoordinator.__init__` saves it to `self.hass`, so it should be available without you needing to save it to `self.hass` in your constructor.

Reply:
I see you removed the assignment in your change so maybe in this comment we talked past each other. Looks right to me now.